docs/userGuide/siteConfiguration.md

```diff
@@ -199,6 +199,18 @@ The example above uses tags as an example of configuring plugin settings, refer
 
 **Specifies that the website should use MarkBind's search functionality.** Default: `true`. See [User Guide: Making the Site Searchable](makingTheSiteSearchable.html) for more details.
 
+#### **`htmlBeautify`**
+
+**Toggles-on html beautification done by [js-beautify](https://github.com/beautify-web/js-beautify) for html files generated by MarkBind.** By default, MarkBind turns this option off to improve the performance of the MarkBind `build` process. 
```

Comment:
I personally lean towards defaulting to having `htmlBeautify` on, and allowing users to turn it off if they really wish to get the performance improvement.

The reason behind this are:
- The build process generally isn't very time critical or performance sensitive for most use cases
- Standardisation of our generated files make it easier to ignore diffs for non-material changes (i.e. markbind might make changes internally that changes the indendentation of files, but these changes don't actually change the markup of the site)
- Having inconsistently formatted files by default isn't really a good look for the product

Reply:
> The build process generally isn't very time critical or performance sensitive for most use cases

Hmm, although there's `markbind serve -o` now, it could still really help when writing the page ( ~20% performance gains ).

> Having inconsistently formatted files by default isn't really a good look for the product

Good point though

> Standardisation of our generated files make it easier to ignore diffs for non-material changes (i.e. markbind might make changes internally that changes the indendentation of files, but these changes don't actually change the markup of the site)

The test sites which don't have `htmlBeautify` turned on are only the `template sites` and `test_site_convert`.

In this sense, how about changing the one for the template sites to be `on` instead? This should solve the problem of presenting a good first impression, while defaulting to `off` if it is not specified.

`test_site_convert` would be an acceptable compromise here I think, but a future pr could always tackle this 😄

Reply:
> Hmm, although there's markbind serve -o now, it could still really help when writing the page ( ~20% performance gains ).

Good point. I wonder what the performance cost for turning beautify on is. Would anyone happen to have some metrics on that?

> In this sense, how about changing the one for the template sites to be on instead? This should solve the problem of presenting a good first impression, while defaulting to off if it is not specified.

I think that the build assets that we generate for the user matters as well. For the typical user, build performance is probably not a big concern. Users who are performance conscious are also generally more willing to tinker with their tools. But this, of course, is difficult to know for certain without further user research :P

I feel that the defaults should be set in a way such that a typical user needs minimal configuration to get an output they are happy with.

Reply:
Maybe as part of user research we can ask @damithc to see if he happens to have any thoughts on this? 😃

Reply:
> Good point. I wonder what the performance cost for turning beautify on is. Would anyone happen to have some metrics on that?

Yup, its in the original issue #1067 

> I think that the build assets that we generate for the user matters as well. For the typical user, build performance is probably not a big concern.

Wouldn't adding `htmlBeautify: on` to the template solve this though? ( since these are the sites the user starts with during `markbind init )

> I feel that the defaults should be set in a way such that a typical user needs minimal configuration to get an output they are happy with.

This does feel cleaner, my worry here is that performance of `markbind serve` also affects ux ultimately though. Not too sure which of the two we should go with either 😬

Reply:
> Yup, its in the original issue #1067

Thanks for pointing me to it :)

> Wouldn't adding htmlBeautify: on to the template solve this though? ( since these are the sites the user starts with during `markbind init )

Ah okay, I see what you mean there. I still think that it makes logical sense for the default behaviour and the default configs we provide to be aligned, but changing the template `site.json` might be an acceptable compromise.

Reply:
@damithc could we get your opinion on whether performance / neatness of generated source files should be prioritised in the default behaviour?

Reply:
Perhaps safer to prioritize neatness as basic users might not need the performance gain (which will be tiny for small web sites) but might get a shock if they to take a look at the generated code? I am OK to go either way though.

Reply:
In that case, we can shift to enabling it by default with an option to turn it off.